src/Microsoft.AspNetCore.SignalR.Client/ConnectionFactory.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.IO.Pipelines;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    // REVIEW: Split into Microsoft.AspNetCore.Sockets.Client?
+    public class ConnectionFactory
```

Comment:
Why do we want the factory? Looking at the code now you now need a couple of factories and a transport to start a connection. Looks like a very complicated API comparing to what we had before.